File: src/jbased/Part6_1.java
package jbased;

import java.util.Arrays;

public class Part6_1 {
    public static void main(String[] args) {
        double[] layerOutputs = {4.8, 1.21, 2.385};
        double[] expValues = Arrays.stream(layerOutputs).map(x -> Math.exp(x)).toArray();
        System.out.println(Arrays.toString(expValues));
        double norm_base = Arrays.stream(expValues).sum();
        expValues = Arrays.stream(expValues).map(x -> x / norm_base).toArray();
        System.out.println(Arrays.toString(expValues));
        System.out.println(Arrays.stream(expValues).sum());
    }
}

File: src/kbased/Part6_1.kt
package kbased
    fun main(args: Array<String>) {
        val layer_outputs = doubleArrayOf(4.8, 1.21, 2.385)

        var expValues = layer_outputs.map { x: Double -> Math.exp(x) }
        println(expValues)
        val normBase = expValues.sum()
        expValues =expValues.map { x: Double -> x / normBase }
        println(expValues)
        println(expValues.sum())
    }